### PR TITLE
Implement `cupy.allclose`

### PR DESCRIPTION
**Pull request by tsurumeso**
This PR implements cupy.allclose.

### CODE REVIEW

File: cupy/logic/comparison.py [hunk 1]
```diff
@@ -17,7 +17,33 @@
...
+            to NaN's in ``b``.
+
+    Returns:
+        bool: if ``True``, two arrays are element-wise equal within a
+            tolerance.
+
+    .. seealso:: :func:`numpy.allclose`
+
+    """
+    res = all(isclose(a, b, rtol=rtol, atol=atol, equal_nan=equal_nan))
```
Comment (toslunar) [inline cupy/logic/comparison.py]: This line should be `res = isclose(...).all()` to support `ndim > 1`.

**Comment by tsurumeso** [comment]
Thank you for your review. I fixed.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
The Jenkins failure (`TestAveragePoolingND_param_0.test_double_backward_cpu`) is irrelevant to the PR.

**Pull request merged**